src/parser.rs

```diff
@@ -181,107 +183,132 @@ impl Decoder for ValueCodec {
     }
 }
 
-pub struct ValueFuture<R> {
-    reader: Option<R>,
-    state: AnySendPartialState,
-    // Intermediate storage for data we know that we need to parse a value but we haven't been able
-    // to parse completely yet
-    remaining: Vec<u8>,
+// https://github.com/tokio-rs/tokio/pull/1687
```

Comment:
I suspect this is not updated with the latest changes from that PR?

Reply:
No, the PR now contains breaking changes so I can't use the new code from there. It would need to wait for a new alpha release of tokio as well.

The unsafe is unfortunate but at least it is just temporary until tokio is updated.